File: CPP/PROGRAM/Laptop.cpp
// KELAS LAPTOP
#pragma once
#include <iostream>
#include <string>
#include "Layar.cpp"
#include "Keyboard.cpp"
#include "Baterai.cpp"

using namespace std;

class Laptop {
    private:
        string nama;
        string jenis;
        Layar layar;
        Keyboard keyboard;
        Baterai baterai;
        string deskripsiMasalah;

    public:
        Laptop() {}
        
        Laptop(string nama, string jenis) {
            this->nama = nama;
            this->jenis = jenis;
        }
        
        void setNama(string nama) {
            this->nama = nama;
        }
        
        void setJenis(string jenis) {
            this->jenis = jenis;
        }
        
        void setLayar(Layar layar) {
            this->layar = layar;
        }
        
        void setKeyboard(Keyboard keyboard) {
            this->keyboard = keyboard;
        }
        
        void setBaterai(Baterai baterai) {
            this->baterai = baterai;
        }
        
        void setDeskripsiMasalah(string deskripsiMasalah) {
            this->deskripsiMasalah = deskripsiMasalah;
        }
        
        string getNama() {
            return nama;
        }
        
        string getJenis() {
            return jenis;
        }
        
        Layar getLayar() {
            return layar;
        }
        
        Keyboard getKeyboard() {
            return keyboard;
        }
        
        Baterai getBaterai() {
            return baterai;
        }
        
        string getDeskripsiMasalah() {
            return deskripsiMasalah;
        }
        
        void tampilkanInfo() {;
            cout << "Nama: " << nama << endl;
            cout << "Jenis: " << jenis << endl;
            cout << "Masalah: " << deskripsiMasalah << endl;
            
            cout << "\n=== LAYAR ===" << endl;
            layar.tampilkanInfo();
            
            cout << "\n=== KEYBOARD ===" << endl;
            keyboard.tampilkanInfo();
            
            cout << "\n=== BATERAI ===" << endl;
            baterai.tampilkanInfo();
        }
        
    ~Laptop() {}
};

File: PYTHON/PROGRAM/Laptop.py
# KELAS LAPTOP
from Layar import Layar
from Keyboard import Keyboard
from Baterai import Baterai

class Laptop:
    def __init__(self, nama="", jenis=""):
        self.__nama = nama
        self.__jenis = jenis
        self.__layar = Layar()
        self.__keyboard = Keyboard()
        self.__baterai = Baterai()
        self.__deskripsi_masalah = ""
    
    def set_nama(self, nama):
        self.__nama = nama
    
    def set_jenis(self, jenis):
        self.__jenis = jenis
    
    def set_layar(self, layar):
        self.__layar = layar
    
    def set_keyboard(self, keyboard):
        self.__keyboard = keyboard
    
    def set_baterai(self, baterai):
        self.__baterai = baterai
    
    def set_deskripsi_masalah(self, deskripsi_masalah):
        self.__deskripsi_masalah = deskripsi_masalah
    
    def get_nama(self):
        return self.__nama
    
    def get_jenis(self):
        return self.__jenis
    
    def get_layar(self):
        return self.__layar
    
    def get_keyboard(self):
        return self.__keyboard
    
    def get_baterai(self):
        return self.__baterai
    
    def get_deskripsi_masalah(self):
        return self.__deskripsi_masalah
    
    def tampilkan_info(self):
        print("Nama:", self.__nama)
        print("Jenis:", self.__jenis)
        print("Masalah:", self.__deskripsi_masalah)
        
        print("\n=== LAYAR ===")
        self.__layar.tampilkan_info()
        
        print("\n=== KEYBOARD ===")
        self.__keyboard.tampilkan_info()
        
        print("\n=== BATERAI ===")
        self.__baterai.tampilkan_info()